ssv/spectest/tests/runner/postconsensus/pre_decided_sc.go

```diff
@@ -0,0 +1,94 @@
+package postconsensus
+
+import (
+	"github.com/attestantio/go-eth2-client/spec"
+
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/ssv"
+	ssvcomparable "github.com/bloxapp/ssv-spec/ssv/spectest/comparable"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+	"github.com/bloxapp/ssv-spec/types/testingutils/comparable"
+)
+
+// preDecidedProposerSC returns state comparison object for the PreDecided Proposer versioned spec test
+func preDecidedProposerSC(version spec.DataVersion) *comparable.StateComparison {
+	ks := testingutils.Testing4SharesSet()
+	cd := testingutils.TestProposerConsensusDataV(version)
+	cdBytes := testingutils.TestProposerConsensusDataBytsV(version)
+
+	return &comparable.StateComparison{
+		ExpectedState: func() types.Root {
+			ret := testingutils.ProposerRunner(ks)
+			ret.GetBaseRunner().State = &ssv.State{
+				StartingDuty: &cd.Duty,
+				PreConsensusContainer: ssvcomparable.SetMessagesInContainer(
+					ssv.NewPartialSigContainer(3),
+					[]*types.SSVMessage{
+						testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[1], ks.Shares[1], 1, 1, version)),
+						testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[2], ks.Shares[2], 2, 2, version)),
+						testingutils.SSVMsgProposer(nil, testingutils.PreConsensusRandaoDifferentSignerMsgV(ks.Shares[3], ks.Shares[3], 3, 3, version)),
+					},
+				),
+				PostConsensusContainer: ssvcomparable.SetMessagesInContainer(
+					ssv.NewPartialSigContainer(3),
+					[]*types.SSVMessage{},
```

Comment:
pretaining to my other comment,
what makes sense to me that this gets filled, but hte node doesn't emit the aggregation until it decides